R/create_profile_matrix.R

```diff
@@ -0,0 +1,243 @@
+# SpatialDecon: mixed cell deconvolution for spatial and/or bulk gene expression
+# data
+# Copyright (C) 2020, NanoString Technologies, Inc.
+#    This program is free software: you can redistribute it and/or modify it
+#    under the terms of the GNU General Public License as published by the Free
+#    Software Foundation, either version 3 of the License, or (at your option)
+#    any later version.
+#    This program is distributed in the hope that it will be useful, but WITHOUT
+#    ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+#    FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+#    more details.
+#    You should have received a copy of the GNU General Public License along
+#    with this program.  If not, see https://www.gnu.org/licenses/.
+# Contact us:
+# NanoString Technologies, Inc.
+# 530 Fairview Avenue N
+# Seattle, WA 98109
+# Tel: (888) 358-6266
+# pdanaher@nanostring.com
+
+
+#' Create Custom Cell Profile Matrix
+#'
+#' Create custom cell profile matrix using single cell data. The average gene expression for each cell type is returned.
+#'
+#' @param mtx cell x gene count matrix
+#' @param cellAnnots cell annotations with cell type and cell name as columns
+#' @param cellTypeCol column containing cell type
+#' @param cellNameCol column containing cell ID/name
+#' @param matrixName name of final profile matrix
+#' @param outDir path to desired output directory
+#' @param geneList gene list to filter profile matrix to 
+#' @param normalize Should data be normalize? (T/F) if TRUE data will be normalize using total gene count
+#' @param scalingFactor what should all values be multiplied by for final matrix, set to 1 if no scaling is wanted
+#' @param minCellNum minimum number of cells of one type needed to create profile, exclusive 
+#' @param minGenes minimum number of genes expressed in a cell, exclusive
+#' @param numCellTypesExpr minimum number of cell types a gene must be expressed in, inclusive
+#' @param filterCellTypes should cell types be filtered for types like mitotic, doublet, low quality, unknown, etc.
```

Comment:
I had trouble interpreting this. Does this mean these cell types will be removed from the final matrix? If so, maybe call it "discardCellTypes", or just explain more clearly.